lib/compiler.js

```diff
@@ -637,9 +659,25 @@ function buildAllMixinInstructions(appRootDir, mixinDirs, options) {
     return meta;
   });
 
+  if (includeMixins.length > 0) {
+    mixins = mixins.filter(function(m) {
+      return (includeMixins.indexOf(m.name) === -1 ? false : true);
+    });
+  }
```

Comment:
I am proposing to move this filtering code out of "loadMixins" to keep "loadMixins" simpler and focused on doing one thing only.